src/test/spec/unified_runner/operation.rs

```diff
@@ -566,10 +568,19 @@ impl TestOperation for Aggregate {
         async move {
             let result = match &self.session {
                 Some(session_id) => {
-                    let entity = test_runner.entities.get(id).unwrap().clone();
+                    enum AggregateEntity {
+                        Collection(Collection<Document>),
+                        Database(Database),
+                        Other(String),
+                    }
+                    let entity = match test_runner.entities.get(id).unwrap() {
```

Comment:
Instead of going through this helper struct can we match on `test_runner.entities.get(id).unwrap()` and clone within the branches? i.e.
```
match test_runner.entities.get(id).unwrap() {
	Entity::Collection(c) => {
		let c = c.clone();
		...
	}
	...
}
```

Reply:
Not easily, unfortunately - `test_runner.entities.get(id).unwrap()` borrows from `test_runner.entities`, so matching on it would do so for the span of the match, and that would conflict with the exclusive borrow needed for `let session = test_runner.get_mut_session(session_id);`.

Reply:
Ah I do remember that being an issue when I implemented this, I think that's why I had that line of code on 569 in the first place.